### PR TITLE
feat(columns): Full-width shiny inputs in column layouts

### PR DESCRIPTION
**Pull request by gadenbuie**
For #727 (for `layout_columns()`)

Sets `.shiny-input-container` to `width: 100%` by default in content areas of `.bslib-grid` layout containers (which are nested one extra layer deep in both `layout_column_wrap()` and `layout_columns()`).

The goal of this PR is to make it much easier to throw inputs into a responsive layout _a la_ `inputPanel()`.

```r
ui <- page_fixed(
  div(class = "my-5"),
  layout_columns(
    sliderInput("lower", "Lower", 0, 100, 10),
    sliderInput("middle", "Middle", 0, 100, 50),
    sliderInput("upper", "Upper", 0, 100, 90),
  )
)

shinyApp(ui, function(input, output, session) {})
```

https://github.com/rstudio/bslib/assets/5420529/ffd1f400-3483-47bc-a4d0-357c4ee7078d

### CODE REVIEW

File: inst/components/scss/grid.scss [hunk 1]
```diff
@@ -3,13 +3,19 @@
   gap: var(--bslib-spacer, 1rem);
   height: var(--bslib-grid-height);
   &.grid {
+    // Don't let intrinsic width of a column affect the width of grid cols
+    grid-template-columns: repeat(var(--bs-columns, 12), minmax(0, 1fr));
```
Comment (cpsievert) [inline inst/components/scss/grid.scss:7]: Might be worth adding a comment that this overrides Bootstrap's `repeat(var(--bs-columns, 12), 1fr);` default

File: inst/components/scss/grid.scss [hunk 2]
```diff
@@ -3,13 +3,19 @@
...
     // For some reason, Bootstrap sets `grid-template-rows: 1fr` by default, which
     // is problematic for a multi-row filling layout. This fixes it...
     // > page_fillable(layout_columns(c(12, 12), plotly::plot_ly(), plotly::plot_ly()))
     grid-template-rows: unset;
     grid-auto-rows: var(--bslib-grid--row-heights);
     @include bslib-breakpoints-css-vars('bslib-grid--row-heights', map-keys($grid-breakpoints));
   }
+
+  & > * > .shiny-input-container {
```
Comment (cpsievert) [inline inst/components/scss/grid.scss:16]: Is the extra `> *` needed here because of the `layout_columns(fillable = T)` thing?

**Review by cpsievert — APPROVED** [review]

File: inst/components/scss/grid.scss (hunk 2 above)
Comment (gadenbuie) [reply inst/components/scss/grid.scss:16]: Yeah `layout_columns()` adds a doc with the bs grid classes and `layout_column_wrap()` puts children in a fillable container. I know the first is always there but I should double check if the second is dependent on the value of `fillable`

Comment (gadenbuie) [reply inst/components/scss/grid.scss:16]: Oh yeah, they both wrap their children in a `grid_item_container()`

**Pull request merged**